### DISCUSSION TITLE
queryString not working?

### DISCUSSION THREAD

**Discussion by chiragparekh** [wire-elements/modal#258] [Q&A]
Hi, I am using modal component, I am trying to implement livewire queryString, so when I set value it sets to the URL query string but when I reopen the modal it returns null value for the property that used for the query string,  ideally it should return the value of the property from the URL.

**Comment by chiragparekh** [comment]
I am using querystring in component like this
![image](https://user-images.githubusercontent.com/1888712/191825535-2b51dc0e-ed82-461f-a5da-60bcbbfb3fe7.png)

so when i use $this->test = '123' it sets the value in the url 123 for the `test` but when I close the modal and reopen the modal instead of returning the 123 it returns null

I have also tried to write like this it doesnt work either

```
public function getQueryString()
{
    return ['test'];
}
```

**Comment by PhiloNL** [comment]
I don't think this is possible; I would recommend passing the query string to your modal component when opening.

**Comment by chiragparekh** [comment]
@PhiloNL thanks for responding, I just wanted to confirm this one for the possibility. I will pass the query string to modal component and fix the issue.
